src/data/fin.lean

```diff
@@ -654,6 +654,11 @@ lemma cast_succ_lt_last (a : fin n) : cast_succ a < last n := lt_iff_coe_lt_coe.
 lemma cast_succ_pos {i : fin (n + 1)} (h : 0 < i) : 0 < cast_succ i :=
 by simpa [lt_iff_coe_lt_coe] using h
 
+@[simp] lemma cast_succ_eq_zero_iff {n : ℕ} (a : fin (n + 1)) : a.cast_succ = 0 ↔ a = 0 := by tidy
```

Comment:
How fast is this proof? Does it make sense to replace it by the output of `by tidy?`. LTE doesn't care too much about speed, but mathlib will want this file to be pretty fast, since it is quite low level.

Reply:
yeah, I did check this; it's not a slow tidy proof, but it's kinda messy:

```
  cases a, split; rintro h; ext1; injections_and_clear; assumption
````

Reply:
so I think keeping `by tidy` is fine

Reply:
```suggestion
@[simp] lemma cast_succ_eq_zero_iff {n : ℕ} (a : fin (n + 1)) : a.cast_succ = 0 ↔ a = 0 :=
⟨λ h, subtype.ext $ (subtype.ext_iff.mp h : _), λ h, h.symm ▸ cast_succ_zero⟩
```

Reply:
Or even
```suggestion
@[simp] lemma cast_succ_eq_zero_iff {n : ℕ} (a : fin (n + 1)) : a.cast_succ = 0 ↔ a = 0 :=
subtype.ext_iff.trans $ (subtype.ext_iff.trans $ by exact iff.rfl).symm
```

Reply:
Shouldn't `(subtype.ext_iff.trans $ by exact iff.rfl).symm` just reduce to `subtype.ext_iff.symm`? Or is this some sort of elaboration order trick?

Reply:
It's an elaboration order trick